### ISSUE TITLE
Validate that Security Requirements are Defined

### ISSUE THREAD

**Issue by dillonredding** [stoplightio/spectral#1091]
**User story.**
As an API designer, when I define a security requirement at the top-level, or for an operation, then I want it to be defined in the `securitySchemes` `components`.

**Is your feature request related to a problem?**
According to the OAS for [security requirement objects](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#securityRequirementObject):

> The name used for each property MUST correspond to a security scheme declared in the [Security Schemes](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#componentsSecuritySchemes) under the [Components Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#componentsObject).

However, spectral raises no errors for the follow OpenAPI doc:

```yaml
openapi: 3.0.2
info:
  title: Foo API
  version: 0.0.0
security:
  - foobar: [] # undefined and therefore invalid
paths: {}
```

**Describe the solution you'd like**
I would like spectral to validate that all declared security requirements are defined as per the OAS requirement.

**Comment by ioggstream** [comment]
Any news on that?

**Comment by ioggstream** [comment]
I stubbed a custom function implementation based on the current oasOpSecurityDefined https://github.com/italia/api-oas-checker/pull/482/files#diff-02327926eb1e6d9b33c3d62bd55da90506c59ad3f913f7431785468be07bd713

Does make sense to you to add something like this in spectral? It is probably an easy feature to add...

**Comment by dillonredding** [comment]
@ioggstream 
Definitely seems like the base/default spectral rules should account for the core OAS.
